src/SFA.DAS.Commitments.Infrastructure/Services/ApprenticeshipEventsPublisher.cs

```diff
@@ -73,7 +73,9 @@ private static Events.Api.Types.ApprenticeshipEvent CreateEvent(Commitment commi
                 TransferSenderId = commitment.TransferSenderId,
                 TransferSenderName = commitment.TransferSenderName,
                 TransferApprovalStatus = (Events.Api.Types.TransferApprovalStatus?) commitment.TransferApprovalStatus,
-                TransferApprovalActionedOn = commitment.TransferApprovalActionedOn
+                TransferApprovalActionedOn = commitment.TransferApprovalActionedOn,
+                StoppedOnDate = apprenticeship.StopDate,
```

Comment:
When editing the stop date, ApprenticeshipEventsPublisher is used to create the event, and the new stop date has already been set in apprenticeship.

Reply:
So this doesn't follow the previous pattern

Reply:
No, there are actually 2 different services for creating and publishing events, ApprenticeshipEvents and ApprenticeshipEventsPublisher (which works in concert with ApprenticeshipEventsList).
Consumers are roughly split 50-50 between the 2 different services (from memory I think the split is something like 5-6).
There are also differences in how the consumers pass the stop and pause date to the services. As stated, when stopping or pausing, the date is passed in effectiveFrom (and any existing stop or pause date is passed in the apprenticeship) and when editing an existing stop date, the apprenticeship already has the new stop date populated.